minidump-processor/src/process_state.rs

```diff
@@ -122,6 +122,87 @@ impl From<MinidumpLinuxProcStatus<'_>> for LinuxProcStatus {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum Limit {
+    Error,
+    Unlimited,
+    Limited(u64)
+}
+
+impl serde::Serialize for Limit {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        match *self {
+            Limit::Error => serializer.serialize_str("err"),
+            Limit::Unlimited => serializer.serialize_str("unlimited"),
+            Limit::Limited(val) => serializer.serialize_u64(val)
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct LinuxProcLimit {
+    pub name: String,
+    pub soft: Limit,
+    pub hard: Limit,
+    pub unit: String
+}
+
+#[derive(Debug, Clone)]
+pub struct LinuxProcLimits {
+    pub limits: HashMap<String, LinuxProcLimit>
+}
+
+fn parse_limit(s: String) -> Limit {
```

Comment:
If you change the parameter to `s: &str` you can change the invocations later from:
```rust
                    soft: parse_limit(m[1].clone()),
                    hard: parse_limit(m[2].clone()),
```
to:
```
                    soft: parse_limit(&m[1]),
                    hard: parse_limit(&m[2]),
```
so you don't have redundant clones.